prow/cmd/deck/static/prow/prow.ts

```diff
@@ -523,18 +523,22 @@ function redraw(fz: FuzzySearch): void {
             r.appendChild(createTextCell(""));
         }
         if (spyglass) {
-            const buildIndex = build.url.indexOf("/build/");
-            if (buildIndex !== -1) {
-                const icon = createIcon("visibility", "View in Spyglass");
-                icon.href = window.location.origin + "/view/gcs/" +
-                    build.url.substring(buildIndex + "/build/".length);
-                const cell = document.createElement("TD");
-                cell.classList.add("icon-cell");
-                cell.appendChild(icon);
-                r.appendChild(cell);
+            if (build.state == 'pending') {
+                let url = window.location.origin + '/view/prowjob/' + build.job + '/' +
+                    build.build_id;
+                r.appendChild(createSpyglassCell(url));
             } else {
-                r.appendChild(createTextCell(""));
+                const buildIndex = build.url.indexOf('/build/');
+                if (buildIndex === -1) {
+                    r.appendChild(createTextCell(''));
+                } else {
```

Comment:
If the build is not pending but hasn't uploaded to GCS yet ... what does the user see?

Reply:
Update: I decided a slightly better solution _was_ in scope for this PR. 7d82fca91008916bd12cbb3eab9a96c00a51ae58